slugify.js

```diff
@@ -53,6 +53,18 @@
         .replace(new RegExp('[\\s' + replacement + ']+', 'g'), replacement)
     }
 
+    if (options.trim) {
+      //escape special characters
+      if('.*+-?^${}()|[]@\\'.includes(replacement)) {
```

Comment:
This assumes that `replacement` is a single character but `slugify()` works just fine with multiple characters for `replacement`. So the `RegExp()` constructor calls below will still have unexpected results if `replacement` is a value like `++` because this `.includes()` will miss values like that. 

This sort of escaping also probably leaves the code open to exponential back-tracking attacks ("RegExp DoS" or "REDoS").

These are increasingly edge cases that might not register for concern if the module had 100 weekly downloads rather than a million. Regardless of the arguable severity of these issues, I think they point to a larger issue, which is that a dynamically generated regular expression is probably not the tool for this job.

This trim behavior is the default behavior of `slug`--I don't even think we have an option to turn it off--and I would love to see `slugify` take on the same behavior because I suspect it's the behavior most/all users want or at least implicitly expect. In `slug`, it's done without regular expressions at all:

1. While walking through each character and replacing with characters from the single- and multi-char maps, if the replacement character is found, then replace it with a space character: https://github.com/Trott/slug/blob/78c5638a5a998223a2b1cb05204e8ae724d1d1ad/slug.js#L195-L197
2. When done replacing characters, run the resulting slug string through `String.prototype.trim()`: https://github.com/Trott/slug/blob/78c5638a5a998223a2b1cb05204e8ae724d1d1ad/slug.js#L208
3. Finally, replace all whitespace characters with the replacement string: https://github.com/Trott/slug/blob/78c5638a5a998223a2b1cb05204e8ae724d1d1ad/slug.js#L209

I haven't looked closely, but I believe this approach would be compatible with `slugify`'s algorithm, and would eliminate the issues around passing unknown strings into `RegExp()`.